brave/src/main/java/brave/sampler/RateLimitingSampler.java

```diff
@@ -0,0 +1,103 @@
+package brave.sampler;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Inspired by on https://github.com/aws/aws-xray-sdk-java/blob/2.0.1/aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/strategy/sampling/reservoir/Reservoir.java
```

Comment:
TODO remove this as it is not really like that impl in any way. the readme should say that the intent is to accept the same parameter as x-ray though.